src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -455,13 +455,13 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
                 {
                     enforceNotNull(returnStatement.Syntax, pendingReturn.State);
                     enforceNotNullWhenForPendingReturn(pendingReturn, expr, returnStatement);
-                    enforceMemberNotNullWhenForPendingReturn(pendingReturn, expr, returnStatement);
+                    enforceMemberNotNullWhenForPendingReturn(pendingReturn, returnStatement);
                 }
             }
 
             return pendingReturns;
 
-            void enforceMemberNotNull(SyntaxNode syntaxOpt, LocalState state)
+            void enforceMemberNotNull(SyntaxNode? syntaxOpt, LocalState state)
```

Comment:
Comment for line 498 below (`if (_symbol is MethodSymbol method)`): I'm surprised we didn't have to declare `method` as nullable.

Reply:
Same comment about line 593 (in local function `makeNotNullMembersMaybeNull`).

Reply:
`if (_symbol is MethodSymbol method)` only succeeds if the `MethodSymbol` is not `null`.

---
In reply to: [463178814](https://github.com/dotnet/roslyn/pull/45640#discussion_r463178814) [](ancestors = 463178814)

Reply:
Aren't we doing a null test on this local below?

Reply:
Ping. Seems like we're missing an annotation because we're going to assign null to `method` (and then we'll test it for null).

Reply:
This is what I'm referring:
```
                if (_symbol is MethodSymbol method)
                {
                    do
                    {
...
                        method = method.OverriddenMethod;
                    }
                    while (method != null);
                }
```

Reply:
I don't think an annotation on `MethodSymbol method` would matter because `OverriddenMethod` is not annotated.

Reply:
What's the correct annotation?